### PR TITLE
feat: utility guides

### PR DESCRIPTION
**Pull request by UndiedGamer**

### CODE REVIEW

File: docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+::: note
+
+If you don't know what a Regex is check the
+[Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)
+
+:::
+
+### Channel Mention Regex
+
+This regex will match any mention of a channel and return the id of the channel
```
Comment (favna) [inline docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx:19]: Add the raw of each regex under each header.

I.e.

---

The raw regex string of this regex is as follows:
```typescript
raw regex here
```

---

No tabs are needed for this. It's the same in everything anyway.

File: docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+larger string.
+
+## Formatted Custom Emoji With Groups
+
+Regex that can capture any animated or non-animated custom Discord emoji. Similar to FormattedCustomEmoji and unlike
+EmojiRegex can also be a substring of a larger string.
+
+### HTTP Regex
+
+This regex matches any url starting with http/https
```
Comment (favna) [inline docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx:50]:
```suggestion
This regex matches any url starting with `http` or `https`
```

File: docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+This regex captures every animated and non-animated emoji's id
+
+### Formatted Custom Emoji Regex
+
+This regex matches any animated or non-animated custom Discord emoji. Unlike EmojiRegex It can be a substring of a
+larger string.
+
+## Formatted Custom Emoji With Groups
+
+Regex that can capture any animated or non-animated custom Discord emoji. Similar to FormattedCustomEmoji and unlike
```
Comment (favna) [inline docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx:45]: When referencing different regexes, make those links.

File: docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+This regex matches any url starting with http/https
+
+### Message Link Regex
+
+This regex can capture the Guild, Channel, and Message ID based on a shareable Discord message link.
+
+### Parsed CustomEmoji Regex
+
+This regex matches any animated or non-animated custom Discord emoji \*without the wrapping `<...>` symbols. This means
```
Comment (favna) [inline docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx:58]: Wild `\*`?

File: docs/Guide/utilities/DiscordJS Utilities/PaginatedMessage.mdx [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+sidebar_position: 0
+title: PaginatedMessage
+---
+
+## PaginatedMessage
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+;
```
Comment (favna) [inline docs/Guide/utilities/DiscordJS Utilities/PaginatedMessage.mdx:11]:
```suggestion
```

File: docs/Guide/utilities/DiscordJS Utilities/PaginatedMessage.mdx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+---
+sidebar_position: 0
+title: PaginatedMessage
+---
+
+## PaginatedMessage
+
+import Tabs from '@theme/Tabs';
```
Comment (favna) [inline docs/Guide/utilities/DiscordJS Utilities/PaginatedMessage.mdx:8]:
```suggestion
---

import Tabs from '@theme/Tabs';
```

Remember, page headers are automatic.

File: docs/Guide/utilities/fetch.mdx [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+title: Using Fetch
+---
+
+## Introduction
+
+Sapphire's fetch package, @sapphire/fetch is a small wrapper around [cross-fetch](https://npmjs.com/package/cross-fetch)
+so it will work in both the web browser and the native node.js environment.
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
```
Comment (favna) [inline docs/Guide/utilities/fetch.mdx:12]: Move to the top

File: docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx [hunk 8]
```diff
@@ -0,0 +1,90 @@
+---
+sidebar_position: 0
+title: Useful Regexes
+---
+
+## Introduction
+
+Sapphire provides a lot of Regexes that you can use in your code
```
Comment (favna) [inline docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx:8]: Add install of @sapphire/discord-utilities

File: docs/Guide/utilities/stopwatch.mdx [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+sidebar_position: 2
+title: Using StopWatch
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Introduction
+
+Provides a set of methods and properties that you can use to accurately measure elapsed time.
```
Comment (favna) [inline docs/Guide/utilities/stopwatch.mdx:11]: This can be a bit more descriptive. I know, I know, you copied it over. That's bad on us too.

**Review by favna — CHANGES_REQUESTED** [review]

File: docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx (hunk 3 above)
Comment (UndiedGamer) [reply docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx:45]: Sure

File: docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx (hunk 4 above)
Comment (UndiedGamer) [reply docs/Guide/utilities/Discord Utilities/UsefulRegexes.mdx:58]: Wild copy pasta

**Review by favna — CHANGES_REQUESTED** [review]
#97 has been merged and this PR has been rebased. Please migrate your `<Tabs` for JS/ESM/TS to the next plugin based syntax

**Comment by UndiedGamer** [comment]
## TODO for now

- [ ]  Add raw regex for reference
- [ ]  Update guides for snowflake

i will re-request a review once i am done with this

**Comment by UndiedGamer** [comment]
@kyranet can you please review the snowflake.mdx file? because you were the one who reworked the package

File: docs/Guide/utilities/snowflake.mdx [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+```typescript ts2esm2cjs
+// Import the Snowflake class
+import { Snowflake } from '@sapphire/snowflake';
+
+// Define a custom epoch const epoch = new Date('2000-01-01T00:00:00.000Z');
+
+// Create an instance of Snowflake const snowflake = new Snowflake(epoch);
+
+// Generate a snowflake with the given epoch const uniqueId = snowflake.generate();
```
Comment (kyranet) [inline docs/Guide/utilities/snowflake.mdx:32]:
```suggestion
// Define a custom epoch
const epoch = new Date('2000-01-01T00:00:00.000Z');

// Create an instance of Snowflake
const snowflake = new Snowflake(epoch);

// Generate a snowflake with the given epoch
const uniqueId = snowflake.generate();
```

File: docs/Guide/utilities/snowflake.mdx [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+// Deconstruct a snowflake with the given epoch
+const uniqueId = snowflake.deconstruct('3971046231244935168');
+```
+
+</details>
+
+:::info
+
+Deconstructing snowflakes is the same for every snowflake
```
Comment (kyranet) [inline docs/Guide/utilities/snowflake.mdx:84]: Should probably explicitly mention `DiscordSnowflake` and `TwitterSnowflake` so they're made aware that they don't have to build a Snowflake instance, but that they can use those 2 exports.

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by favna — CHANGES_REQUESTED** [review]
no spaces in paths allowed 

image


Use underscores and then if you want it to show up as `Discord Utilities` on the sidebar add a `category.yml`

File: docs/Guide/utilities/Discord_Utilities/UsefulRegexes.mdx [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+## Introduction
+
+Sapphire provides a lot of Regexes that you can use in your code
+
+::: note
+
+If you don't know what a Regex is check the
+[Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)
+
+:::
```
Comment (favna) [inline docs/Guide/utilities/Discord_Utilities/UsefulRegexes.mdx:15]:
```suggestion
:::note

If you don't know what a Regex is check the
[Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions)

:::
```

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**